packages/tools/src/lib/build-cache/github/GitHubBuildCache.ts

```diff
@@ -1,116 +1,92 @@
-      );
-      await promptForGitHubToken();
+  }
+
+  async getRepoDetails() {
+    if (!this.repoDetails) {
+      this.repoDetails = await detectGitHubRepoDetails();
     }
     return this.repoDetails;
   }
 
-  async query({
+  async list({
     artifactName,
+    limit,
   }: {
-    artifactName: string;
-  }): Promise<RemoteArtifact | null> {
-    const repoDetails = await this.detectRepoDetails();
-    if (!getGitHubToken()) {
-      logger.warn(`No GitHub Personal Access Token found.`);
-      return null;
-    }
-
-    if (!repoDetails) {
-      return null;
-    }
-
+    artifactName?: string;
+    limit?: number;
+  }): Promise<RemoteArtifact[]> {
+    const repoDetails = await this.getRepoDetails();
     const artifacts = await fetchGitHubArtifactsByName(
       artifactName,
-      repoDetails
+      repoDetails,
+      limit
     );
+    return artifacts.map((artifact) => ({
+      name: artifact.name,
+      url: artifact.downloadUrl,
+    }));
+  }
+
+  async download({
+    artifactName,
+  }: {
+    artifactName: string;
+  }): Promise<Response> {
+    const repoDetails = await this.getRepoDetails();
+    const artifacts = await this.list({ artifactName });
     if (artifacts.length === 0) {
-      return null;
+      throw new RnefError(`No artifact found with name "${artifactName}"`);
     }
-
-    return {
-      name: artifacts[0].name,
-      downloadUrl: artifacts[0].downloadUrl,
-    };
+    return fetch(artifacts[0].url, {
+      headers: {
+        Authorization: `token ${repoDetails.token}`,
+        'Accept-Encoding': 'None',
+      },
+    });
   }
 
-  async download({
-    artifact,
-    loader,
+  async delete({
+    artifactName,
   }: {
-    artifact: RemoteArtifact;
-    loader: ReturnType<typeof spinner>;
-  }): Promise<LocalArtifact> {
-    const artifactPath = getLocalArtifactPath(artifact.name);
-    await downloadGitHubArtifact(
-      artifact.downloadUrl,
-      artifactPath,
-      this.name,
-      loader
+    artifactName: string;
+  }): Promise<RemoteArtifact[]> {
+    const repoDetails = await this.getRepoDetails();
+    const artifacts = await fetchGitHubArtifactsByName(
+      artifactName,
+      repoDetails,
+      undefined
     );
-    await extractArtifactTarballIfNeeded(artifactPath);
-    return {
-      name: artifact.name,
-      path: artifactPath,
-    };
+    if (artifacts.length === 0) {
+      throw new RnefError(`No artifact found with name "${artifactName}"`);
+    }
+    return await deleteGitHubArtifacts(artifacts, repoDetails, artifactName);
   }
-}
 
-async function extractArtifactTarballIfNeeded(artifactPath: string) {
-  const tarPath = path.join(artifactPath, 'app.tar.gz');
-
-  // If the tarball is not found, it means the artifact is already unpacked.
-  if (!fs.existsSync(tarPath)) {
-    return;
+  async upload(): Promise<RemoteArtifact> {
+    throw new RnefError('Uploading artifacts to GitHub is not supported.');
   }
```

Comment:
would be really nice to point to more technical reason here. maybe there's some github doc?